solr/contrib/ltr/src/test/org/apache/solr/ltr/ranking/TestModelQuery.java

```diff
@@ -147,15 +147,17 @@ public void testModelQueryEquality() throws ModelException {
     assertEquals(m1, m2);
     assertEquals(m1.hashCode(), m2.hashCode());
 
-    final HashMap<String,String> externalFeatureInfo = new HashMap<>();
-    externalFeatureInfo.put("queryIntent", "company");
+    final HashMap<String,String[]> externalFeatureInfo = new HashMap<>();
+    externalFeatureInfo.put("queryIntent", new String[] {"company"});
+    externalFeatureInfo.put("user_query", new String[] {"abc"});
     m2.setExternalFeatureInfo(externalFeatureInfo);
 
     assertFalse(m1.equals(m2));
     assertFalse(m1.hashCode() == m2.hashCode());
 
-    final HashMap<String,String> externalFeatureInfo2 = new HashMap<>();
-    externalFeatureInfo2.put("queryIntent", "company");
+    final HashMap<String,String[]> externalFeatureInfo2 = new HashMap<>();
+    externalFeatureInfo2.put("user_query", new String[] {"abc"});
+    externalFeatureInfo2.put("queryIntent", new String[] {"company"});
```

Comment:
Should assert that the m1 & m2 hashcode() are equal too.  The assertequals call as far as I know only calls the equals function